src/Fixer/Whitespace/NoSpacesAroundOffsetFixer.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\ConfigurationException\InvalidFixerConfigurationException;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Javier Spagnoletti <phansys@gmail.com>
+ */
+final class NoSpacesAroundOffsetFixer extends AbstractFixer
+{
+    /**
+     * @var string[]
+     */
+    private $configuration = array();
+
+    /**
+     * Default target/configuration.
+     *
+     * @var string[]
+     */
+    private static $defaultConfiguration = array(
+        'inside',
+        'outside',
+    );
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        if (null === $configuration) {
+            $this->configuration = self::$defaultConfiguration;
+
+            return;
+        }
+
+        foreach ($configuration as $name) {
+            if (!in_array($name, self::$defaultConfiguration, true)) {
+                throw new InvalidFixerConfigurationException($this->getName(), sprintf('Unknown configuration option "%s".', $name));
+            }
+        }
+
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound(array('[', CT_ARRAY_INDEX_CURLY_BRACE_OPEN));
```

Comment:
If the user doesn't provide configuration I would expect the default to be used, but if the user provides an empty (or invalid) configuration I would expect a configuration exception to be thrown.
But we can go both ways (since it is an edge case anyway) :)